### PR TITLE
Jenkings configuration

### PR DESCRIPTION
**Pull request by antongulenko**
Added Jenkinsfile for tests, coverage and docker container building. Fixed pom for tests.

### CODE REVIEW

**Comment by antongulenko** [comment]
When merging this PR, please select "squash commit" to remove the useless intermediate commits.

File: Jenkinsfile [hunk 1]
```diff
@@ -9,6 +9,11 @@ pipeline {
         docker_image = 'teambitflow/bitflow4j'
     }
     stages {
+        stage('SonarQube analysis') {
+            withSonarQubeEnv('CIT SonarQube') {
```
Comment (antongulenko) [inline Jenkinsfile]: Pls move this stage to after tests

**Comment by srnbckr** [comment]
I'm still testing some stuff. Please do not merge yet.

File: Jenkinsfile (hunk 1 above)
Comment (srnbckr) [reply Jenkinsfile]: done

**Comment by srnbckr** [comment]
I'm done. Ready to merge from my side.

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        }
+        stage("Quality Gate") {
+            steps {
+                timeout(time: 30, unit: 'MINUTES') {
+                    waitForQualityGate abortPipeline: true
+                }
+            }
+        }
+        stage('Build container') {
+            //when {
```
Comment (flohannes) [inline Jenkinsfile]: remove those outcomment lines pleeeease :)

**Review by flohannes — CHANGES_REQUESTED** [review]

**Review by flohannes — APPROVED** [review]

**Pull request merged**